docs/tests/general/core/dev_tests.rst

```diff
@@ -0,0 +1,309 @@
+    3. Click the **Forget** button located at the bottom-right of the device icon.
+        - **Expected result:** Initially, the device is disconnected and then successfully removed from the device browser.
+
+**Result:** PASS/FAIL
+
+Test 4: Edit the name of the device.
+------------------------------------
+
+**UID:** TST.DEV.EDIT_NAME
+
+**Description:** This test checks if the **Edit pen** button work correctly.
+
+**Test prerequisites:**
+    - :ref:`TST.ADD.PLUGINS_ADD_SINGLE <tst-add-plugins-add>`.
+
+**Preconditions:**
+    - Scopy is installed on the system.
+    - iio-emu is installed on the system.
+    - Use :ref:`M2k.Emu <m2k-emu-dev>` setup.
+    - OS: Windows, Linux-x86_64, Linux-arm64, Linux-arm32, macOS.
+
+**Steps:**
+    1. Place the cursor in the top-right corner of the device icon in the device browser.
+    2. After the pen icon appears, click on it.
+    3. Delete "M2k".
+    4. Write "M3k".
+    5. Press enter. 
+
+Expected result: The device name changes both in the device browser and in the tool menu.
+
+**Result:** PASS/FAIL
+
+Test 5: Select a tool. 
+----------------------
+
+**UID:** TST.DEV.SELECT_TOOL
+
+**Description:** This test checks if the tool infrastructure works correctly.
+
+**Test prerequisites:**
+    - :ref:`TST.CONN.SUCC <tst-conn-succ>`.
+
+**Preconditions:**
+    - Scopy is installed on the system.
+    - iio-emu is installed on the system.
+    - Use :ref:`M2k.Emu <m2k-emu-dev>` setup.
+    - OS: Windows, Linux-x86_64, Linux-arm64, Linux-arm32, macOS.
+
+**Steps:**
+    1. Click on the device icon from device browser.
+    2. Click on **Connect** button.
+    3. Click on "Oscilloscope" tool.
+        - **Expected result:** The "Oscilloscope" tool is displayed and the tool menu item is highlighted.
+    4. Click on "Home" menu entry.
+        - **Expected result:** The device page is displayed and the home menu item is highlighted.
+
+**Result:** PASS/FAIL
+
+.. _tst-dev-detach-tool:
+
+Test 6: Detach a tool.
+----------------------
+
+**UID:** TST.DEV.DETACH_TOOL
+
+**Description:** This test checks if the tool detachment mechanism is working correctly.
+
+**Test prerequisites:**
+    - :ref:`TST.CONN.SUCC <tst-conn-succ>`.
+
+**Preconditions:**
+    - Scopy is installed on the system.
+    - iio-emu is installed on the system.
+    - Use :ref:`M2k.Emu <m2k-emu-dev>` setup.
+    - OS: Windows, Linux-x86_64, Linux-arm64, Linux-arm32, macOS.
+
+**Steps:**
+    1. Click on the device icon from device browser.
+    2. Click on **Connect** button.
+    3. Double click on "Oscilloscope" tool.
+        - **Expected result:** The "Oscilloscope" tool is integrated into a separate window from the application. The window title should be: "Scopy-M2kPlugin-Oscilloscope-ip:127.0.0.1".
+    4. Possible steps.
+        1. Close the "Oscilloscope" window.
+            - **Expected result:** The "Oscilloscope" tool is reintegrated into the main window. 
+        2. Close the  application.
```

Comment:
I would rephrase this and add 2 more steps here. The tester shouldn't choose from 2 possible scenarios. The test should have a single result (pass or fail).
I would go the following way:
4. close the osc window + expected result
5. detach it again without an expected result
6. close the application + expected result